atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1914,9 +1919,12 @@ private void commitWrites(TransactionService transactionService) {
                 timedAndTraced(
                         "markingTransactionInProgress", () -> transactionService.markInProgress(getStartTimestamp()));
 
-                // Write to the targeted sweep queue. We must do this before writing to the key value service -
-                // otherwise we may have hanging values that targeted sweep won't know about.
-                timedAndTraced("writingToSweepQueue", () -> sweepQueue.enqueue(writesByTable, getStartTimestamp()));
+                // Optimistically write to the targeted sweep queue.
+                // Whilst ideally must do this before writing to the key value service - to avoid hanging values that
+                // targeted sweep won't know about - this is relatively expensive, and so err on the side of
+                // running this concurrently with other operations (and accepting sometimes we'll miss some stuff).
+                Future<?> writeToSweepQueueFuture = writeToSweepQueueExecutor.submit(() -> timedAndTraced(
```

Comment:
Unfortunately, with transactions4 this would become a correctness issue. It would be possible, on transactions 4:

1. write to the KVS
2. write to the sweep queue gets stuck in GC
3. we lose the immutable TS lock
4. sweep makes progress -- because there were no entries for this transaction we don't mark it as abandoned
5. reads believe this transaction was committed

It may be possible to still do this in parallel, if we modify the way we mark transactions as known committed -- if we at that point read through the transactions table and make sure we mark anything still in progress or explicitly aborted as abandoned I think that would work. @sudiksha27 thoughts?